Mathlib/SetTheory/Cardinal/Basic.lean

```diff
@@ -750,6 +750,24 @@ theorem le_sum {ι} (f : ι → Cardinal) (i) : f i ≤ sum f := by
 theorem mk_sigma {ι} (f : ι → Type*) : #(Σ i, f i) = sum fun i => #(f i) :=
   mk_congr <| Equiv.sigmaCongrRight fun _ => outMkEquiv.symm
 
+theorem mk_sigma_congr {ι ι' : Type u} {f : ι → Type v} {g : ι' → Type v} (e : ι ≃ ι')
+    (h : ∀ i, #(f i) = #(g (e i))) : #(Σ i, f i) = #(Σ i, g i) :=
+  mk_congr <| Equiv.sigmaCongr e fun i ↦ Classical.choice <| Cardinal.eq.mp (h i)
+
+theorem mk_sigma_congr' {ι : Type u} {ι' : Type v} {f : ι → Type max w (max u v)}
+    {g : ι' → Type max w (max u v)} (e : ι ≃ ι')
+    (h : ∀ i, #(f i) = #(g (e i))) : #(Σ i, f i) = #(Σ i, g i) :=
+  mk_congr <| Equiv.sigmaCongr e fun i ↦ Classical.choice <| Cardinal.eq.mp (h i)
+
+theorem mk_sigma_congrRight {ι : Type u} {f g : ι → Type v} (h : ∀ i, #(f i) = #(g i)) :
+    #(Σ i, f i) = #(Σ i, g i) :=
+  mk_sigma_congr (Equiv.refl ι) h
+
+theorem mk_sigma_congrRight_prop {ι : Type u} {f g : ι → Type v} {p : ι → Prop}
+    (h : ∀ i, p i → #(f i) = #(g i)) : #(Σ i, p i → f i) = #(Σ i, p i → g i) := mk_congr <|
```

Comment:
This is not the right statement; "subtypes" of sigmas are spelt as one of:
```lean
#check Σ x : Nat, Σ' h : x > 0, Fin x
#check Σ x : Nat, (x > 0) ×' Fin x
#check Σ x : Nat, PLift (x > 0) × Fin x
```
This is a bit like the common binder confusion with `Exists`:
```lean
∃ x : { x // x > 0}, x.val^2 = 4
∃ x, x > 0 ∧ x^2 = 4 -- equivalent to the above
∃ x, x > 0 → x^2 = 4 -- meaningless, you can use ` x=0`.
```